modules/ingester/instance.go

```diff
@@ -140,9 +161,6 @@ func (i *instance) Push(ctx context.Context, req *tempopb.PushRequest) error {
 		return status.Errorf(codes.FailedPrecondition, "%s max live traces exceeded for tenant %s: %v", overrides.ErrorPrefixLiveTracesExceeded, i.instanceID, err)
 	}
 
-	i.tracesMtx.Lock()
-	defer i.tracesMtx.Unlock()
-
 	id, err := pushRequestTraceID(req)
```

Comment:
i wonder if we should just delete `Push()`? it hasn't been used for 3 or 4 releases.

Reply:
Agree but didn't seem in-scope for this PR.